src/core/utils/parse_cmdline.hpp

```diff
@@ -180,10 +180,7 @@ otError ParseAsBool(const char *aString, bool &aBool);
  * @retval kErrorInvalidArgs  The string does not contain valid IPv6 address.
  *
  */
-inline otError ParseAsIp6Address(const char *aString, otIp6Address &aAddress)
-{
-    return otIp6AddressFromString(aString, &aAddress);
-}
+otError ParseAsIp6Address(const char *aString, otIp6Address &aAddress);
```

Comment:
Ideally, do what results in smallest code size. General rule of thumb is single line should be inline.

Reply:
I agree... I think we should aim for option giving us smaller code size...

I think the obvious case is simple getter/setter methods where we access just a member vraible. 

All other cases, I would say it may better to check the impact for code size.. I have been surprised in past by cases where I expected `inline`ing a simple method to help with code size and it did the reverse.